<a href="https://colab.research.google.com/github/ehaabbasil/pgpapi/blob/master/%23TEAM36%20Data%20Challenge%20Escooter%20Level_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline 

#main_dir_usama = '/content/drive/MyDrive/University of Maryland/Spring_2021/Data_Lv3_UMDOTS_Escooters.csv'
ehaab = '/content/drive/MyDrive/TEAM36/Data_Lv3_UMDOTS_Escooters.csv'
pd.__version__

'1.1.5'

# New section

In [ ]:
#df = pd.read_csv(main_dir_usama)
df = pd.read_csv(ehaab)

In [ ]:
df.head()

Unnamed: 0  Unnamed: 0.1  ...   END LONG    END LAT
0           0             0  ... -76.934285  38.990739
1           1             1  ... -76.949165  38.992849
2           2             2  ... -76.933335  38.992676
3           3             3  ... -76.943546  38.982476
4           4             4  ... -76.943257  38.986101

[5 rows x 16 columns]

In [ ]:
df.columns

Index(['Unnamed: 0', 'Unnamed: 0.1', 'CREATED', 'START', 'END', 'DISTANCE',
       'RIDE_ID', 'CUSTOMER_ID', 'MINUTES', 'VEHICLE_TYPE', 'PATH',
       'TIMESTAMPS', 'START LONG', 'START LAT', 'END LONG', 'END LAT'],
      dtype='object')

In [ ]:
df.dtypes

Unnamed: 0        int64
Unnamed: 0.1      int64
CREATED          object
START            object
END              object
DISTANCE        float64
RIDE_ID           int64
CUSTOMER_ID       int64
MINUTES           int64
VEHICLE_TYPE     object
PATH             object
TIMESTAMPS       object
START LONG      float64
START LAT       float64
END LONG        float64
END LAT         float64
dtype: object

In [ ]:
del df['Unnamed: 0']

In [ ]:
del df['Unnamed: 0.1']

In [ ]:
df.head()

CREATED                      START  ...   END LONG    END LAT
0  2020-10-03 03:32:15+00:00  2020-10-03 03:32:15+00:00  ... -76.934285  38.990739
1  2020-10-03 03:07:39+00:00  2020-10-03 03:07:39+00:00  ... -76.949165  38.992849
2  2020-10-03 03:17:05+00:00  2020-10-03 03:17:05+00:00  ... -76.933335  38.992676
3  2020-10-03 02:24:58+00:00  2020-10-03 02:24:58+00:00  ... -76.943546  38.982476
4  2020-10-03 03:19:34+00:00  2020-10-03 03:19:34+00:00  ... -76.943257  38.986101

[5 rows x 14 columns]

In [ ]:
#deleted first two unnecessary columns

In [ ]:
#now if i want to seperately keep columns 

In [ ]:
s=df[['START LAT','START LONG']]
print(s)
x=s.value_counts()
x
#maximum a coordinate has occured in start lat & start long

       START LAT  START LONG
0      38.990656  -76.934163
1      38.982087  -76.938181
2      38.979613  -76.937600
3      38.982470  -76.943553
4      38.982084  -76.938116
...          ...         ...
40320  38.988110  -76.943936
40321  38.986389  -76.935917
40322  38.996377  -76.940231
40323  38.991956  -76.944153
40324  38.990645  -76.934167

[40325 rows x 2 columns]


START LAT  START LONG
38.992447  -76.949093    20
38.992699  -76.935208    17
38.988723  -76.940666    14
38.982561  -76.943550     7
38.984513  -76.941714     5
                         ..
38.991116  -76.942908     1
38.991114  -76.933588     1
           -76.946452     1
38.991113  -76.932574     1
38.955716  -76.939235     1
Length: 40039, dtype: int64

In [ ]:
c=df[['END LAT','END LONG']]
print(c)
z=c.value_counts()
z

         END LAT   END LONG
0      38.990739 -76.934285
1      38.992849 -76.949165
2      38.992676 -76.933335
3      38.982476 -76.943546
4      38.986101 -76.943257
...          ...        ...
40320  38.992061 -76.934104
40321  38.992196 -76.933818
40322  38.992842 -76.945600
40323  38.987199 -76.940216
40324  38.982452 -76.936088

[40325 rows x 2 columns]


END LAT    END LONG  
38.992447  -76.949093    20
38.992699  -76.935208    17
38.988723  -76.940666    14
38.982561  -76.943550     7
38.984513  -76.941714     5
                         ..
38.991137  -76.948636     1
38.991136  -76.933331     1
38.991134  -76.934034     1
38.991132  -76.934333     1
38.904981  -77.042210     1
Length: 40153, dtype: int64

In [ ]:
df.iloc[0,1]

'2020-10-03 03:32:15+00:00'

In [ ]:
df.iloc[:1] #this gives the first row

CREATED                      START  ...   END LONG    END LAT
0  2020-10-03 03:32:15+00:00  2020-10-03 03:32:15+00:00  ... -76.934285  38.990739

[1 rows x 14 columns]

In [ ]:
df.iloc[0:]

CREATED  ...    END LAT
0      2020-10-03 03:32:15+00:00  ...  38.990739
1      2020-10-03 03:07:39+00:00  ...  38.992849
2      2020-10-03 03:17:05+00:00  ...  38.992676
3      2020-10-03 02:24:58+00:00  ...  38.982476
4      2020-10-03 03:19:34+00:00  ...  38.986101
...                          ...  ...        ...
40320  2020-10-24 22:21:40+00:00  ...  38.992061
40321  2020-10-24 21:28:43+00:00  ...  38.992196
40322  2020-10-25 01:22:00+00:00  ...  38.992842
40323  2020-10-25 00:55:24+00:00  ...  38.987199
40324  2020-10-24 06:03:05+00:00  ...  38.982452

[40325 rows x 14 columns]

In [ ]:
#.unique() tells the unique values in column
#.nununique tells non unique in each column

In [ ]:
start=['START LONG','START LAT']

In [ ]:
df.iloc[3:4]

CREATED                      START  ...   END LONG    END LAT
3  2020-10-03 02:24:58+00:00  2020-10-03 02:24:58+00:00  ... -76.943546  38.982476

[1 rows x 14 columns]

In [ ]:
df.columns

Index(['CREATED', 'START', 'END', 'DISTANCE', 'RIDE_ID', 'CUSTOMER_ID',
       'MINUTES', 'VEHICLE_TYPE', 'PATH', 'TIMESTAMPS', 'START LONG',
       'START LAT', 'END LONG', 'END LAT'],
      dtype='object')

In [ ]:
pth=df['PATH']

In [ ]:
import numpy as np

In [ ]:
df[["DISTANCE"]].mean(axis=1)

0        0.008
1        1.062
2        0.847
3        3.447
4        0.386
         ...  
40320    0.794
40321    0.404
40322    0.629
40323    0.567
40324    5.220
Length: 40325, dtype: float64

In [ ]:
df.describe()

DISTANCE       RIDE_ID  ...      END LONG       END LAT
count  38052.000000  4.032500e+04  ...  40325.000000  40325.000000
mean      13.959499  2.451930e+06  ...    -76.940179     38.987819
std      295.521959  8.616648e+05  ...      0.006851      0.006926
min        0.000000  1.316030e+06  ...    -77.421865     38.904981
25%        0.430000  1.538282e+06  ...    -76.944944     38.982469
50%        0.708000  3.137506e+06  ...    -76.940803     38.988514
75%        0.984000  3.248853e+06  ...    -76.936047     38.992561
max    12623.444000  3.339713e+06  ...    -76.723846     39.528800

[8 rows x 8 columns]

In [ ]:
#! conda install ipyleaflet

In [ ]:
#! conda install -c conda-forge ipyleaflet

In [ ]:
# !pip install geopandas
# ! pip install ipyleaflet

import folium
from shapely.geometry import Point 

from shapely.geometry import Polygon

%matplotlib inline




In [ ]:
center = [38.9897, -76.9378]

m = folium.Map(location=center, zoom_start=13, tiles="Stamen Terrain")

folium.Marker(
    location=center,
    popup="College Park",
    icon=folium.Icon(icon="info-sign"),
).add_to(m)

folium.Circle(
    radius=1000,
    location=center,
    popup="College Park Area",
    color="crimson",
    fill=False,
).add_to(m)

m.save("index.html")
m

MOST COMMON STARTING AND ENDING POINTS FROM DATA SET IS ELKTON HALL IN NORTH CAMPUS AT UMD
[38.992447  -76.949093]

In [ ]:
# ! jupyter nbextension install --py --symlink --sys-prefix ipyleaflet
# ! jupyter nbextension enable --py --sys-prefix ipyleaflet

# !jupyter labextension install @jupyter-widgets/jupyterlab-manager
# !jupyter labextension install jupyter-leaflet

# ! jupyter labextension install js



# import ipyleaflet
# from ipyleaflet import Map
# from ipyleaflet import (Map,DrawControl)

# m=Map(center=[38.990656,-76.934163],zoom=10)
# d=DrawControl(circle={'shapeOptions':{'color':'#0000FF'}})
# m.add_control(d)
# m
# display(m)
#not displaying map



# from ipyleaflet import Map, Marker
# center = (52.204793, 360.121558)
# m = Map(center=center, zoom=15)
# marker = Marker(location=center, draggable=True)
# m.add_layer(marker);
# display(m)
# m.save('map.html',title='test')
# type(m)


In [ ]:
#df and #np
pth.head()

0    {\n  "coordinates": [\n    [\n      -76.934163...
1    {\n  "coordinates": [\n    [\n      -76.938180...
2    {\n  "coordinates": [\n    [\n      -76.937599...
3    {\n  "coordinates": [\n    [\n      -76.943553...
4    {\n  "coordinates": [\n    [\n      -76.938115...
Name: PATH, dtype: object

In [ ]:
# october 2020 
df.iloc[0:]

CREATED  ...    END LAT
0      2020-10-03 03:32:15+00:00  ...  38.990739
1      2020-10-03 03:07:39+00:00  ...  38.992849
2      2020-10-03 03:17:05+00:00  ...  38.992676
3      2020-10-03 02:24:58+00:00  ...  38.982476
4      2020-10-03 03:19:34+00:00  ...  38.986101
...                          ...  ...        ...
40320  2020-10-24 22:21:40+00:00  ...  38.992061
40321  2020-10-24 21:28:43+00:00  ...  38.992196
40322  2020-10-25 01:22:00+00:00  ...  38.992842
40323  2020-10-25 00:55:24+00:00  ...  38.987199
40324  2020-10-24 06:03:05+00:00  ...  38.982452

[40325 rows x 14 columns]

In [ ]:
paths = df.Path.tolist()


NameError: ignored

In [ ]:

from datetime import datetime, date, timedelta, timezone
datatime=df['CREATED'].drop(37903)



In [ ]:
t=df['CREATED'].str.contains('2020', na = False)

In [ ]:
#FILTERING TO october 2020 only 
oct20=df[(df['START']>'2020-10-00')& (df['END']<'2020-10-31')]

In [ ]:
# now find average distance 
oct20.describe()

DISTANCE       RIDE_ID  ...       END LAT  created
count  21106.00000  2.112700e+04  ...  21127.000000  21127.0
mean      22.42555  3.233527e+06  ...     38.987720      1.0
std      377.71228  6.269384e+04  ...      0.006192      0.0
min        0.00000  3.117521e+06  ...     38.939232      1.0
25%        0.40500  3.181882e+06  ...     38.982448      1.0
50%        0.70500  3.237741e+06  ...     38.988637      1.0
75%        0.97100  3.289324e+06  ...     38.992351      1.0
max    12623.44400  3.332332e+06  ...     39.056681      1.0

[8 rows x 9 columns]

In [ ]:
s=oct20[['START LAT','START LONG']]
print(s)
x=s.value_counts()
x

       START LAT  START LONG
0      38.990656  -76.934163
1      38.982087  -76.938181
2      38.979613  -76.937600
3      38.982470  -76.943553
4      38.982084  -76.938116
...          ...         ...
40320  38.988110  -76.943936
40321  38.986389  -76.935917
40322  38.996377  -76.940231
40323  38.991956  -76.944153
40324  38.990645  -76.934167

[21127 rows x 2 columns]


START LAT  START LONG
38.992447  -76.949093    20
38.992699  -76.935208    17
38.988723  -76.940666    14
38.984513  -76.941714     4
38.991582  -76.946516     3
                         ..
38.991574  -76.946841     1
           -76.942957     1
38.991575  -76.946494     1
38.991576  -76.946821     1
38.962263  -76.934364     1
Length: 21012, dtype: int64

In [ ]:
c=oct20[['END LAT','END LONG']]
print(c)
v=c.value_counts()
v

         END LAT   END LONG
0      38.990739 -76.934285
1      38.992849 -76.949165
2      38.992676 -76.933335
3      38.982476 -76.943546
4      38.986101 -76.943257
...          ...        ...
40320  38.992061 -76.934104
40321  38.992196 -76.933818
40322  38.992842 -76.945600
40323  38.987199 -76.940216
40324  38.982452 -76.936088

[21127 rows x 2 columns]


END LAT    END LONG  
38.992447  -76.949093    20
38.992699  -76.935208    17
38.988723  -76.940666    14
38.984513  -76.941714     4
38.976563  -76.937690     3
                         ..
38.991567  -76.947016     1
           -76.946543     1
38.991569  -76.946816     1
           -76.946544     1
38.939232  -76.946893     1
Length: 21051, dtype: int64

In [ ]:
center = [38.992699, -76.93520]

m = folium.Map(location=center, zoom_start=13, tiles="Stamen Terrain")

folium.Marker(
    location=center,
    popup="College Park",
    icon=folium.Icon(icon="info-sign"),
).add_to(m)

folium.Circle(
    radius=150,
    location=center,
    popup="College Park Area",
    color="crimson",
    fill=False,
).add_to(m)

m.save("index.html")
m


0        38.990656
1        38.982087
2        38.979613
3        38.982470
4        38.982084
           ...    
40320    38.988110
40321    38.986389
40322    38.996377
40323    38.991956
40324    38.990645
Name: START LAT, Length: 40325, dtype: float64

In [ ]:
start

['START LONG', 'START LAT']

In [ ]:
df[start]

START LONG  START LAT
0      -76.934163  38.990656
1      -76.938181  38.982087
2      -76.937600  38.979613
3      -76.943553  38.982470
4      -76.938116  38.982084
...           ...        ...
40320  -76.943936  38.988110
40321  -76.935917  38.986389
40322  -76.940231  38.996377
40323  -76.944153  38.991956
40324  -76.934167  38.990645

[40325 rows x 2 columns]

In [ ]:
oct19=df[(df['START']>'2019-10-00')& (df['END']<'2019-10-31')]

In [ ]:
oct19.describe()

DISTANCE       RIDE_ID  ...       END LAT  created
count  15481.000000  1.765700e+04  ...  17657.000000  17657.0
mean       3.667393  1.508957e+06  ...     38.987949      1.0
std      141.222457  8.974555e+04  ...      0.007801      0.0
min        0.006000  1.316030e+06  ...     38.904981      1.0
25%        0.474000  1.445706e+06  ...     38.982486      1.0
50%        0.715000  1.520190e+06  ...     38.987744      1.0
75%        1.006000  1.589707e+06  ...     38.992902      1.0
max    11812.447000  1.642427e+06  ...     39.528800      1.0

[8 rows x 9 columns]

In [ ]:
c=oct19[['END LAT','END LONG']]
print(c)
v=c.value_counts()
v

         END LAT   END LONG
239    38.992995 -76.933048
240    38.993253 -76.946110
241    39.002287 -76.942077
242    38.993162 -76.946191
243    39.001886 -76.941957
...          ...        ...
38381  38.993187 -76.946178
38382  38.993269 -76.946144
38383  38.990593 -76.934028
38384  38.978320 -76.929246
38385  38.990621 -76.934171

[17657 rows x 2 columns]


END LAT    END LONG  
38.982561  -76.943550    7
38.982575  -76.943728    4
38.982432  -76.943613    3
38.993066  -76.946279    3
38.979034  -76.939941    3
                        ..
38.990676  -76.934017    1
           -76.934163    1
           -76.934159    1
38.990675  -76.934050    1
38.904981  -77.042210    1
Length: 17567, dtype: int64

In [ ]:
s=oct19[['START LAT','START LONG']]
print(s)
x=s.value_counts()
x

       START LAT  START LONG
239    38.990585  -76.934334
240    38.985183  -76.942697
241    38.990022  -76.940788
242    38.981604  -76.937635
243    38.993068  -76.945771
...          ...         ...
38381  38.982444  -76.943574
38382  38.980939  -76.937579
38383  38.978389  -76.927651
38384  38.978320  -76.929246
38385  38.982431  -76.943563

[17657 rows x 2 columns]


START LAT  START LONG
38.982561  -76.943550    7
38.982432  -76.943613    4
38.982575  -76.943728    4
38.989981  -76.940823    3
38.990700  -76.934128    3
                        ..
38.990678  -76.934035    1
           -76.934087    1
           -76.933962    1
38.990677  -76.934169    1
38.955716  -76.939235    1
Length: 17499, dtype: int64

In [ ]:
center = [38.989981, -76.940823]

m = folium.Map(location=center, zoom_start=13, tiles="Stamen Terrain")

folium.Marker(
    location=center,
    popup="College Park",
    icon=folium.Icon(icon="info-sign"),
).add_to(m)

folium.Circle(
    radius=150,
    location=center,
    popup="College Park Area",
    color="crimson",
    fill=False,
).add_to(m)

m.save("index.html")
m

In [ ]:
#